utils/economic-checks/src/util.rs

```diff
@@ -0,0 +1,342 @@
+    let initial_accounts_num = 1 + (rng.next_u32() % 1000); // [1..1000]
+    let mut accounts = vec![(root_acc.clone(), 1_000_000_000_000_000_u128)];
+    for _ in 1..initial_accounts_num {
+        let mut acc_id = [0_u8; 32];
+        rng.fill_bytes(&mut acc_id);
+        let balance = (rng.next_u64() >> 14) as u128; // approx. up to 10^15
+        accounts.push((acc_id.into(), balance));
+    }
+    accounts
+}
+
+pub(crate) fn new_test_ext(
+    balances: Vec<(impl Into<AccountId32>, u128)>,
+    initial_authorities: Vec<(AccountId32, AuraId, GrandpaId)>,
+    root_key: AccountId32,
+) -> sp_io::TestExternalities {
+    let mut t = system::GenesisConfig::default()
+        .build_storage::<Runtime>()
+        .unwrap();
+
+    pallet_balances::GenesisConfig::<Runtime> {
+        balances: balances
+            .into_iter()
+            .map(|(acc, balance)| (acc.into(), balance))
+            .chain(
+                initial_authorities
+                    .iter()
+                    .cloned()
+                    .map(|(acc, _, _)| (acc, 1000)),
+            )
+            .collect(),
+    }
+    .assimilate_storage(&mut t)
+    .unwrap();
+
+    AuraConfig {
+        authorities: initial_authorities
+            .iter()
+            .cloned()
+            .map(|(_, x, _)| x)
+            .collect(),
+    }
+    .assimilate_storage(&mut t)
+    .unwrap();
+
+    BasicExternalities::execute_with_storage(&mut t, || {
+        <GrandpaConfig as GenesisBuild<Runtime>>::build(&GrandpaConfig {
+            authorities: initial_authorities
+                .iter()
+                .map(|x| (x.2.clone(), 1))
+                .collect(),
+        });
+        // pallet_grandpa::Pallet::<Runtime>::initialize(
+        //     &initial_authorities
+        //         .iter()
+        //         .map(|x| (x.2.clone(), 1))
+        //         .collect(),
+        // );
+
+        <TransactionPaymentConfig as GenesisBuild<Runtime>>::build(
+            &TransactionPaymentConfig::default(),
+        );
+    }); // .unwrap();
+
+    SudoConfig {
+        key: Some(root_key),
+    }
+    .assimilate_storage(&mut t)
+    .unwrap();
+
+    let mut ext: sp_io::TestExternalities = t.into(); //= sp_io::TestExternalities::new(t);
+    ext.execute_with(|| System::set_block_number(1));
+    ext
+}
+
+pub(crate) fn with_offchain_ext(
+    balances: Vec<(impl Into<AccountId32>, u128)>,
+    initial_authorities: Vec<(AccountId32, AuraId, GrandpaId)>,
+    root_key: AccountId32,
+) -> (sp_io::TestExternalities, Arc<RwLock<PoolState>>) {
+    let mut ext = new_test_ext(balances, initial_authorities, root_key);
+    let (offchain, _) = TestOffchainExt::new();
+    let (pool, pool_state) = TestTransactionPoolExt::new();
+
+    ext.register_extension(OffchainDbExt::new(offchain.clone()));
+    ext.register_extension(OffchainWorkerExt::new(offchain));
+    ext.register_extension(TransactionPoolExt::new(pool));
+
+    (ext, pool_state)
+}
+
+#[allow(dead_code)]
```

Comment:
```suggestion
```
Why allowing dead code?

if it optionally used or only for tests there are `#[allow(unused)]` or `#[cfg(test)]`

Reply:
fuzzer is essentially a test, is it not? )
Anyway, I agree that we don't have to keep any dead code, hence this macro is useless